includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/publish/PublishConfigurationPlugin.kt

```diff
@@ -3,37 +3,117 @@ package com.badoo.reaktive.publish
-        target.tasks.register(TASK_FILTERED_PUBLISH_TO_MAVEN_LOCAL) {
-            dependsOn(project.tasks.matching { it is AbstractPublishToMaven && it.enabled })
+    private fun setupPublishing(project: Project) {
+        val publishing = project.extensions.getByType(PublishingExtension::class.java)
+        publishing.repositories {
+            maven {
+                name = "sonatype"
+                val repositoryId = project.findProperty("sonatype.repository")
+                url = URI.create(
+                    if (repositoryId != null) {
+                        "https://oss.sonatype.org/service/local/staging/deployByRepositoryId/$repositoryId/"
+                    } else {
+                        "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+                    }
+                )
+                credentials {
+                    username = project.findProperty("sonatype.username").toString()
+                    password = project.findProperty("sonatype.password").toString()
+                }
+            }
+        }
+        publishing.publications.withType(MavenPublication::class.java).configureEach {
+            pom {
+                name.set(project.name)
+                description.set(project.description)
+                url.set("https://github.com/badoo/Reaktive")
+                licenses {
+                    license {
+                        name.set("Apache-2.0")
+                        url.set("https://github.com/badoo/Reaktive/blob/master/LICENSE")
+                    }
+                }
+                scm {
+                    connection.set("scm:git:ssh://github.com/badoo/Reaktive.git")
+                    developerConnection.set("scm:git:ssh://github.com/badoo/Reaktive.git")
+                    url.set("https://github.com/badoo/Reaktive")
+                }
+            }
+        }
+    }
+
+    private fun setupSign(project: Project) {
+        val inMemoryKey = project.findProperty("signing.key")?.toString()
+        // Constants from PgpSignatoryFactory.PROPERTIES
+        val password = project.findProperty("signing.password")?.toString()
+        val keyFile = project.findProperty("signing.secretKeyRingFile")?.toString()
+        val keyId = project.findProperty("signing.keyId")?.toString()
+
+        if (inMemoryKey == null && keyFile == null) {
+            project.logger.warn("No signing config provided, skip signing")
+            return
+        }
+        project.plugins.apply(SigningPlugin::class.java)
+        project.extensions.getByType(SigningExtension::class.java).apply {
+            isRequired = true
+            sign(
+                project.extensions.getByType(PublishingExtension::class.java).publications
+            )
+            if (inMemoryKey != null) {
+                useInMemoryPgpKeys(inMemoryKey, password)
+            }
+        }
+    }
+
+    private fun createFilteredPublishTasks(project: Project) {
```

Comment:
Why do we need the additional task? Is it for local publications?

Reply:
Because otherwise we will invoke compilation of jvm/js/linux on macOS host by invoking default `publishToSonatype` directly.